### PR TITLE
collector integration tests: run every test with both bulk and by-sub…

### PR DESCRIPTION
**Pull request by jgraff2**
…ject collection, track request counts

Changes:
- run all tests twice, once for bulk collection and once for by-subject collection.
- track number of requests made to the mocked endpoints, so we can confirm the collector doesn't make redundant logins.

Corresponding collector changes:
https://github.com/salesforce/refocus-collector/pull/186

Tests won't pass in travis until collector is published.

### CODE REVIEW

File: tests/collector/integration.js [hunk 1]
```diff
@@ -212,1238 +276,1559 @@ const nockConfig = [
...
-            .then(() => u.postGenerator(gen2))
-            .then(() => u.expectHeartbeatGens(coll2, { added: [gen2] })),
-        ))
-
-        .then(() => Promise.join(
-          u.expectBulkUpsertSamples(coll1, '60s', ['sub1|asp1']),
-          u.expectBulkUpsertSamples(coll2, '60s', ['sub2|asp2']),
-        ))
-      );
+        it('only gets token once, uses it for all subsequent requests', () => {
```
Comment (jgraff2) [inline tests/collector/integration.js]: @pallavi2209 this test verifies that the collector doesn't send multiple login requests.
I confirmed that this fails with the old version of the collector (before your fix), with a login count of 6.

**Comment by iamigo** [comment]
@jgraff2 should this change also update package.json to set minimum collector version to 1.2.1?

**Comment by jgraff2** [comment]
Yeah, I always forget to do that, thanks for the reminder.

File: tests/collector/integration.js (hunk 1 above)
Comment (pallavi2209) [reply tests/collector/integration.js]: wow - these 3 tests! Thank you for the effort @jgraff2

**Pull request merged**